libminifi/src/core/Processor.cpp

```diff
@@ -223,7 +223,18 @@ bool Processor::flowFilesQueued() {
   return false;
 }
 
-bool Processor::flowFilesOutGoingFull() {
+uint64_t Processor::numIncomingFlowFilesQueued() const {
+  std::lock_guard<std::mutex> lock{mutex_};
+
+  uint64_t sum = 0;
+  for (const auto& connectable : _incomingConnections) {
+    auto connection = std::static_pointer_cast<Connection>(connectable);
```

Comment:
Does this assume that on a connectable, every incoming connection is actually a `Connection`?  If so, why not change the type of the `_incomingConnections` to a `std::set<std::shared_ptr<Connection>>`?

Reply:
It does seem like we always store Connections in it, so this is a good idea, although not related to this change.